frontend/appflowy_tauri/src/appflowy_app/components/_shared/DatabaseSort/DatabaseSortItem.tsx

```diff
@@ -0,0 +1,205 @@
+      onSave({ fieldId: currentFieldId, sort: currentOrder });
+    }
+  }, [currentFieldId, currentOrder]);
+
+  const onSelectFieldClick = (id: string) => {
+    setCurrentFieldId(id);
+    // set ascending order by default
+    setCurrentOrder('asc');
+    setShowFieldSelect(false);
+  };
+
+  const onFieldClick: MouseEventHandler = () => {
+    if (!refFieldSelect.current) return;
+    const { top, left, height } = refFieldSelect.current.getBoundingClientRect();
+
+    setFieldSelectTop(top + height);
+    setFieldSelectLeft(left);
+    setShowFieldSelect(true);
+  };
+
+  const onSelectOrderClick = (order: 'asc' | 'desc') => {
+    setCurrentOrder(order);
+    setShowOrderSelect(false);
+  };
+
+  const onOrderClick: MouseEventHandler = () => {
+    if (!refOrderSelect.current) return;
+    const { top, left, height } = refOrderSelect.current.getBoundingClientRect();
+
+    setOrderSelectTop(top + height);
+    setOrderSelectLeft(left);
+    setShowOrderSelect(true);
+  };
+
+  return (
+    <>
+      <div className={'flex items-center gap-2'}>
+        <button className={'flex-shrink-0 rounded p-1 hover:bg-fill-list-hover'}>
+          <i className={'block h-[16px] w-[16px]'}>
+            <DragElementSvg></DragElementSvg>
+          </i>
+        </button>
+        <div className={'flex flex-1 items-center gap-2'}>
+          <div
+            className={`flex w-[180px] items-center justify-between rounded-lg border px-2 py-1 ${
+              showFieldSelect ? 'border-fill-hover' : 'border-shade-4'
+            }`}
+            ref={refFieldSelect}
+            onClick={onFieldClick}
+          >
+            {currentFieldId ? (
+              <div className={'flex items-center gap-2'}>
+                <i className={'block h-5 w-5'}>
+                  <FieldTypeIcon fieldType={fields[currentFieldId].fieldType}></FieldTypeIcon>
+                </i>
+                <span>{fields[currentFieldId].title}</span>
+              </div>
+            ) : (
+              <span className={'text-shade-4'}>Select a field</span>
+            )}
+          </div>
+          <div
+            className={`flex w-[180px] items-center justify-between rounded-lg border px-2 py-1 ${
+              showOrderSelect ? 'border-fill-hover' : 'border-shade-4'
+            }`}
+            ref={refOrderSelect}
+            onClick={onOrderClick}
+          >
+            {currentOrder ? (
+              <SortLabel value={currentOrder}></SortLabel>
+            ) : (
+              <span className={'text-shade-4'}>Select order</span>
+            )}
+          </div>
+        </div>
+        <button
+          onClick={() => onDelete?.()}
+          className={`rounded p-1 hover:bg-fill-list-hover ${data ? 'opacity-100' : 'opacity-0'}`}
+        >
+          <i className={'block h-[16px] w-[16px]'}>
+            <TrashSvg />
+          </i>
+        </button>
+      </div>
+      {showFieldSelect && (
+        <PopupSelect
```

Comment:
Maybe you can use import { Select } from @mui/material

Reply:
using ButtonPopoverList